module/os/freebsd/zfs/zfs_vfsops.c

```diff
@@ -1023,6 +1026,16 @@ zfsvfs_setup(zfsvfs_t *zfsvfs, boolean_t mounting)
 			zfsvfs->z_vfs->vfs_flag &= ~VFS_RDONLY;
 		} else {
 			dsl_dir_t *dd;
+			zap_stats_t zs;
+
+			if (zap_get_stats(zfsvfs->z_os, zfsvfs->z_unlinkedobj,
+			    &zs) == 0) {
+				dataset_kstats_update_nunlinks_kstat(
+				    &zfsvfs->z_kstat, zs.zs_num_entries);
+			}
+			dprintf_ds(zfsvfs->z_os->os_dsl_dataset,
+			    "num_entries in unlinked set: %llu",
+			    zs.zs_num_entries);
```

Comment:
`zs` will be in an undefined state here if `zap_get_stats` failed. I take it this debug message will be going away?

Reply:
Or move it to inside the if block.

Reply:
This was just copied from the original patch. The same bug seems to exist in the os/linux version still: https://github.com/openzfs/zfs/blob/master/module/os/linux/zfs/zfs_vfsops.c#L882

Do we think it makes most sense to just move the dprintf into the if?

Reply:
Well, that's what I said ;).

Reply:
@alek-p what say you? Is this printf still useful?